### PR TITLE
Add LetsEncrypt certificate support

### PR DESCRIPTION
**Pull request by zquestz**
Adds new flags to automate LetsEncrypt certificates!

### CODE REVIEW

File: cmd/main.go [hunk 1]
```diff
@@ -78,12 +84,50 @@ func performCommand(cmd *cobra.Command, args []string) error {
...
+	if config.AutoCert != "" {
+		m := &autocert.Manager{
+			Cache:      autocert.DirCache(certDir),
+			Prompt:     autocert.AcceptTOS,
+			HostPolicy: autocert.HostWhitelist(config.AutoCert),
+		}
+
+		go http.ListenAndServe(":http", m.HTTPHandler(nil))
+
+		return m, err
```
Comment (eyeamera) [inline cmd/main.go]: Nit: `return m, nil` - `err` should always be nil here

Comment (zquestz) [reply cmd/main.go]: Fixed.

File: cmd/main.go [hunk 2]
```diff
@@ -78,12 +84,50 @@ func performCommand(cmd *cobra.Command, args []string) error {
...
+			HostPolicy: autocert.HostWhitelist(config.AutoCert),
+		}
+
+		go http.ListenAndServe(":http", m.HTTPHandler(nil))
+
+		return m, nil
 	}
 
-	return server.Start(config.Port, config.Cert, config.Key, t, config.Debug)
+	return nil, nil
```
Comment (eyeamera) [inline cmd/main.go]: Minor nit: I tend to prefer early returns. Having the check above be:

```go
if config.AutoCert == "" {
    return nil, nil
}
...
return m, nil
```

might read better

Comment (zquestz) [reply cmd/main.go]: Agreed and fixed.

**Pull request merged**